ts/src/abstractMachine/JSWriter.ts

```diff
@@ -33,8 +33,11 @@ export default class JSWriter implements AbstractMachine {
     // @ts-ignore
     private logger : MyLogger = new MyLogger(J$.initParams.outputFile);
 
+    // the codeArr represents a StringBuilder for the output file so an I/O operation doesn't occur whenever a callback is referenced.
+    private codeArr : string[] = [];
+
```

Comment:
Great optimization! I'm sure this is already much faster than synchronously writing to the file during each instruction like Augur did before.